### PR TITLE
Early support for mathematical content in Kindle for PC.

### PR DESCRIPTION
**Pull request by jcsteh**
### Link to issue number:
None.

### Summary of the issue:
A future version of Kindle will expose MathML for books which contain it. We can thus provide access to math in those books using our existing math support.

### Description of how this pull request fixes the issue:
Kindle implementation:

1. A future version of Kindle will expose MathML on embedded objects with ROLE_SYSTEM_EQUATION via a "mathml" IAccessible2 object attribute.
2. Alternatively, if there is no MathML, alt text might be exposed instead via accName.
3. Both MathML and alt text can be exposed, in which case we should always use the MathML.

In this PR:

1. Update the Kindle app module to fetch MathML accordingly. It also ensures that alt text (which is placed in the "content" ControlField attribute) doesn't get reported if there is MathML.
2. MozillaCompoundTextInfo NVDAObjectAtStart property: Return the embedded object for graphics, math, etc. instead of the parent document. This is necessary to allow for interaction with math.
3. BrowseModeTreeInterceptor: When activating an NVDAObject, if doAction isn't implemented, just ignore it rather than throwing an exception. Otherwise, pressing enter on a graphic (which can't be activated in Kindle) throws an exception.
4. Update the User Guide with details about this, as well as adding a link to the MathPlayer documentation about navigation commands.

### Testing performed:
Tested with builds and test cases provided by Amazon. Unfortunately, neither builds nor content are publicly available yet.

### Known issues with pull request:
No known issues with the implementation. One concern is that content probably won't be publicly available before NVDA 2017.4 is released, so this won't get any public testing before release. However, we don't want to wait until 2018.1 for this to be available, so we're announcing it as "early support".

### Change log entry:
New Features:

```
- Early support for reading and interactive navigation of mathematical content for Kindle books with accessible math. (#7536)
```

### CODE REVIEW

File: source/appModules/kindle.py [hunk 1]
```diff
@@ -271,6 +278,21 @@ def updateSelection(self):
 		log.debug("Setting selection to (%d, %d)" % (sel._startOffset, sel._endOffset))
 		sel.updateSelection()
 
+	def _getControlFieldForObject(self, obj, ignoreEditableText=True):
+		field = super(BookPageViewTextInfo, self)._getControlFieldForObject(obj, ignoreEditableText=ignoreEditableText)
+		if field and field["role"] == controlTypes.ROLE_MATH:
+			try:
+				field["mathMl"] = obj.mathMl
+			except LookupError:
+				mathMl = None
```
Comment (michaelDCurran) [inline source/appModules/kindle.py]: Does this variable get used?

**Review by michaelDCurran — APPROVED** [review]
Should the what's new entry mention the word Kindle perhaps?

**Comment by jcsteh** [comment]
> Should the what's new entry mention the word Kindle perhaps?

Haha. My bad. I'll fix that.

**Comment by codeofdusk** [comment]
Are there now test cases for this? [these](https://www.amazon.com/How-Prove-Structured-Daniel-Velleman-ebook/dp/B07TBM9LW6/ref=sr_1_1?keywords=how+to+prove+it+3&qid=1585528562&sr=8-1) [books](https://www.amazon.com/Algorithm-Design-Manual-Steven-Skiena-ebook/dp/B00B8139Z8/ref=sr_1_2?keywords=algorithm+design+manual&qid=1585528616&s=digital-text&sr=1-2), for example, don't seem to have accessible mathematics...

**Pull request merged**